delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/GraphResourceToNTriples.scala

```diff
@@ -19,16 +19,17 @@ object GraphResourceToNTriples extends Pipe {
   override def inType: Typeable[GraphResource] = Typeable[GraphResource]
   override def outType: Typeable[NTriples]     = Typeable[NTriples]
 
+  def graphToNTriples(graphResource: GraphResource): Task[NTriples] = {
+    val graph = graphResource.graph ++ graphResource.metadataGraph
+    Task.fromEither(graph.toNTriples)
+  }
+
   override def apply(element: SuccessElem[GraphResource]): Task[Elem[NTriples]] = {
-    val graph = element.value.graph ++ element.value.metadataGraph
-    Task
-      .fromEither(graph.toNTriples)
-      .redeem(
-        err => element.failed(err),
-        {
-          case ntriples if ntriples.isEmpty => element.dropped
-          case ntriples                     => element.success(ntriples)
-        }
-      )
+    element
+      .evalMap(graphToNTriples)
+      .map {
+        case ntriples: SuccessElem[NTriples] if ntriples.value.isEmpty => element.dropped
```

Comment:
This part should also be in `graphToNTriples` to have a consistent behaviour when using the pipe or not.